packages/x-data-grid/src/components/GridSkeletonLoadingOverlay.tsx

```diff
@@ -52,220 +52,244 @@ const useUtilityClasses = (ownerState: OwnerState) => {
 
 const getColIndex = (el: HTMLElement) => parseInt(el.getAttribute('data-colindex')!, 10);
 
-const GridSkeletonLoadingOverlay = forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
-  function GridSkeletonLoadingOverlay(props, forwardedRef) {
-    const rootProps = useGridRootProps();
-    const { slots } = rootProps;
-    const isRtl = useRtl();
-    const classes = useUtilityClasses({ classes: rootProps.classes });
-    const ref = React.useRef<HTMLDivElement>(null);
-    const handleRef = useForkRef(ref, forwardedRef);
-    const apiRef = useGridApiContext();
-    const dimensions = useGridSelector(apiRef, gridDimensionsSelector);
-    const viewportHeight = dimensions?.viewportInnerSize.height ?? 0;
-    const skeletonRowsCount = Math.ceil(viewportHeight / dimensions.rowHeight);
-    const totalWidth = useGridSelector(apiRef, gridColumnsTotalWidthSelector);
-    const positions = useGridSelector(apiRef, gridColumnPositionsSelector);
-    const inViewportCount = React.useMemo(
-      () => positions.filter((value) => value <= totalWidth).length,
-      [totalWidth, positions],
-    );
-    const allVisibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
-    const columns = React.useMemo(
-      () => allVisibleColumns.slice(0, inViewportCount),
-      [allVisibleColumns, inViewportCount],
-    );
-    const pinnedColumns = useGridSelector(apiRef, gridVisiblePinnedColumnDefinitionsSelector);
+export const GridSkeletonLoadingOverlayInner = forwardRef<
```

Comment:
There's an issue with scrolling not being in sync:

https://github.com/user-attachments/assets/e1ab930b-90f0-4451-9196-813bdad163c5

Works ok on the [skeleton overlay demo](https://deploy-preview-16282--material-ui-x.netlify.app/x/react-data-grid/overlays/#skeleton):

https://github.com/user-attachments/assets/87504ebe-75a8-4a4d-adbc-27d88fa73268

Reply:
Looks like a broader issue, seems like we need to explicitly pass the `slotProps.loadingOverlay.noRowsVariant` for the scroll to be in sync.

Reported a separate [issue](https://github.com/mui/mui-x/issues/16436) and a potential fix. Thanks for reporting.